scala-exercises-client/src/main/scala/scripts/ExercisesJS.scala

```diff
@@ -1,17 +1,83 @@
 package scripts
 
+import config.Routes
+import messages.EvaluationRequest
+import org.scalajs.dom.ext.{ AjaxException, Ajax }
+import common.ExtAjax._
+import upickle._
 import scala.scalajs.js
 import org.scalajs.dom
-import org.scalajs.jquery.{ jQuery ⇒ $ }
+import org.scalajs.jquery.{ jQuery ⇒ $, JQuery }
+import scala.scalajs.concurrent.JSExecutionContext.Implicits.queue
 
 object ExercisesJS extends js.JSApp {
 
+  val resAssert = """(?s)\((res[0-9]*)\)""".r
+
   def main(): Unit = {
+    insertInputs
+
+  }
+
+  private def insertInputs = {
+    $("pre code").each({ (_: Any, code: dom.Element) ⇒
+      val lineWithInput = resAssert.replaceAllIn($(code).text(), """(<input type="text" data-res="$1"/>)""")
+      $(code).html(lineWithInput)
+    })
     runHighlight
   }
 
-  def runHighlight = $("pre code").each({ (_: Any, code: dom.Element) ⇒
-    js.Dynamic.global.hljs.highlightBlock(code)
-  })
+  private def runHighlight = {
+    $("pre code").each({ (_: Any, code: dom.Element) ⇒
+      js.Dynamic.global.hljs.highlightBlock(code)
+    })
+    activeInputs
+  }
+
+  private def activeInputs = {
+    $(".exercise-code>input").each({ (_: Any, input: dom.Element) ⇒
+      $(input)
+        .width(inputSize($(input).value().toString.length))
+        .keyup((e: dom.Event) ⇒ $(input).width(inputSize($(input).value().toString.length)))
+        .blur((e: dom.Event) ⇒ evalAnswered($(input).closest(".exercise")))
+    })
+  }
+
+  def evalAnswered(exercise: JQuery) = {
+    val inputs = exercise.find("input")
+    val filledInputs = inputs.filter({ (index: Int, e: dom.Element) ⇒ isAnswered(e) })
+
+    if (filledInputs.length == inputs.length) evalExercise(exercise)
+  }
+
+  def evalExercise(exercise: JQuery) = {
+    val section = exercise.attr("data-section")
```

Comment:
Can these be processed in a monadic context such as `Option` such as it won't fail if param is found?